src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1463,6 +1463,61 @@ bool ONNXConstantOp::inferShapes() {
   return true;
 }
 
+//===----------------------------------------------------------------------===//
+// Concat
+
+bool ONNXConcatOp::inferShapes() {
+  printf("hi alex\n");
+  int inputNum = getNumOperands();
+  for (int i = 0; i < inputNum; ++i) {
+    if (!getOperand(i).getType().cast<RankedTensorType>()) {
+      emitError("Input tensor(s) not ranked");
+      return false;
+    }
+  }
+  // Checking value of axis parameter/
+  auto commonType = getOperand(0).getType().cast<RankedTensorType>();
+  auto commonShape = commonType.getShape();
+  auto commonRank = commonShape.size();
+  auto axisIndex = axis().getSExtValue();
+  if (!(axisIndex >= 0 && axisIndex < commonRank)) {
+    emitError("Concat axis value out of bound");
+    return false;
+  }
+  // Initial cummlative size is that of the first operand.
+  int cummulativeAxisSize = commonShape[axisIndex];
```

Comment:
Are we planning to handle the dynamic case at all?

Reply:
good point, restricted to static dimensions along the reduction for now